staging_vespalib/src/vespa/vespalib/metrics/simple_metrics_manager.h

```diff
@@ -71,17 +69,20 @@ class SimpleMetricsManager : public MetricsManager
     Dimension dimension(const vespalib::string &name) override;
     Label label(const vespalib::string &value) override;
     PointBuilder pointBuilder(Point from) override;
-    Point pointFrom(PointMapBacking &&map) override;
+    Point pointFrom(PointMap::BackingMap map) override;
     Snapshot snapshot() override;
+    Snapshot totalSnapshot() override;
 
     // for use from Counter only
-    void add(CounterIncrement inc) override {
+    void add(Counter::Increment inc) override {
         _currentBucket.add(inc);
     }
     // for use from Gauge only
-    void sample(GaugeMeasurement value) override {
+    void sample(Gauge::Measurement value) override {
         _currentBucket.sample(value);
     }
+
+    void tick() { collectCurrentBucket(); }
```

Comment:
for testing I think we should either pass the current time as parameter here or have a polymorphic time 'service' passed to the constructor. Also consider putting the ticker thread outside this class.